src/cli-util.js

```diff
@@ -392,9 +326,188 @@ function formatFiles(argv, filepatterns) {
   });
 }
 
+function createUsage() {
+  const options = constant.detailedOptions;
+  // Add --no-foo after --foo.
+  const optionsWithOpposites = options.map(option => [
+    option.description ? option : null,
+    option.oppositeDescription
+      ? Object.assign({}, option, {
+          name: `no-${option.name}`,
+          type: "boolean",
+          description: option.oppositeDescription
+        })
+      : null
+  ]);
+  const flattenedOptions = [].concat
+    .apply([], optionsWithOpposites)
+    .filter(Boolean);
+
+  const groupedOptions = groupBy(flattenedOptions, option => option.category);
+
+  const firstCategories = [
+    constant.CATEGORY_OUTPUT,
+    constant.CATEGORY_FORMAT,
+    constant.CATEGORY_CONFIG,
+    constant.CATEGORY_EDITOR
+  ];
+  const lastCategories = [constant.CATEGORY_OTHER];
+  const restCategories = Object.keys(groupedOptions).filter(
+    category =>
+      firstCategories.indexOf(category) === -1 &&
+      lastCategories.indexOf(category) === -1
+  );
+  const allCategories = firstCategories.concat(restCategories, lastCategories);
```

Comment:
Do we really need this block? Can't we just do:

```js
const categoryOrder = [
  constant.CATEGORY_OUTPUT,
  constant.CATEGORY_FORMAT,
  constant.CATEGORY_CONFIG,
  constant.CATEGORY_EDITOR,
  constant.CATEGORY_OTHER
];
```

in cli-constant.js and be done with it? What do you think, @ikatyang?

Reply:
My original thought is that we might forget to edit the ordering in cli-util.js, cause unordered category missing thus the snapshot tests passed with no change. We have code view to catch that kind of things, but I'm not sure if we will remember to 😅. Any thoughts?

Reply:
Yeah, I guess it makes sense to keep this to help us from making mistakes in the future. It's not so many lines after all.